### PR TITLE
Component audit - Modify/Delete API OS tests

### PR DESCRIPTION
**Pull request by shubhamsg199**
This PR aims to refactor Operating System API tests. There were many OS tests which are combined into one end-to-end test.
Removing  tests which are covered already either in e2e or in some other tests

### CODE REVIEW

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_positive_end_to_end_os

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_positive_end_to_end_os

**Review by adarshdubey-star — APPROVED** [review]
Ack

File: tests/foreman/api/test_provisioning.py [hunk 1]
```diff
@@ -187,126 +185,125 @@ def test_verify_bugzilla_1114640(self, target_sat):
 class TestOperatingSystem:
     """Tests for operating systems."""
 
-    @pytest.mark.tier1
-    @pytest.mark.parametrize('name', **parametrized(valid_data_list()))
-    def test_positive_create_with_name(self, name, target_sat):
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_provisioning.py:192]: Can you keep this parametrized test for OS create with names from valid_data_list ?

**Review by synkd — APPROVED** [review]

File: tests/foreman/api/test_provisioning.py (hunk 1 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_provisioning.py:192]: Updated.

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_positive_end_to_end_os

File: tests/foreman/api/test_provisioning.py [hunk 2]
```diff
@@ -187,126 +186,142 @@ def test_verify_bugzilla_1114640(self, target_sat):
...
-            given architecture.
-
-        :CaseLevel: Integration
-        """
-        arch = target_sat.api.Architecture().create()
-        operating_sys = target_sat.api.OperatingSystem(architecture=[arch]).create()
-        assert len(operating_sys.architecture) == 1
-        assert operating_sys.architecture[0].id == arch.id
+        os = target_sat.api.OperatingSystem(name=name).create()
+        assert os.name == name
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_provisioning.py:324]: Can we add finalizer step here for the teardown of deleting OS cretated and in other tests creating OS?

File: tests/foreman/api/test_provisioning.py [hunk 3]
```diff
@@ -187,126 +186,142 @@ def test_verify_bugzilla_1114640(self, target_sat):
 class TestOperatingSystem:
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_provisioning.py:186]: Can we either move this OS tests into tests/foreman/api/test_operatingsystem.py or OS CLI tests into provisioning to keep this tests consistent?

File: tests/foreman/api/test_provisioning.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_provisioning.py:324]: We can but the OS is created in different orgs which won't affect other OS tests. same we followed in OS cli tests and we do have deleting OS scenario covered.

File: tests/foreman/api/test_provisioning.py (hunk 3 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_provisioning.py:186]: I think moving it into `tests/foreman/api/test_operatingsystem.py` will be good

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_operatingsystem.py

**Pull request merged**